File: patch_shell.js
const fs = require('fs');
const path = require('path');

const filesToPatch = [
    "cuti.html",
    "pelanggaran.html",
    "dashboard/index.html",
    "tip.html",
    "peminjaman.html",
    "peminjaman_admin.html",
    "karyawan.html",
    "users.html"
];

filesToPatch.forEach(file => {
    if (!fs.existsSync(file)) {
        console.log(`File not found: ${file}`);
        return;
    }
    
    let content = fs.readFileSync(file, 'utf8');

    const prefix = file.includes('/') ? '../' : '';
    const baseDir = file.includes('/') ? '..' : '.';
    
    const cssLinks = `  <link rel="stylesheet" href="${prefix}shared/design-tokens.css">\n  <link rel="stylesheet" href="${prefix}shared/shell.css">\n  <style>`;
    if (!content.includes(`<link rel="stylesheet" href="${prefix}shared/design-tokens.css">`)) {
        content = content.replace('<style>', cssLinks);
    }
        
    const jsLink = `  <script src="${prefix}shared/shell.js" data-base="${baseDir}"></script>\n</body>`;
    
    // Check if it's already there
    if (!content.includes(`src="${prefix}shared/shell.js"`)) {
        // Find the LAST occurrence of </body>
        const lastBodyIdx = content.lastIndexOf('</body>');
        if (lastBodyIdx !== -1) {
            content = content.substring(0, lastBodyIdx) + jsLink + content.substring(lastBodyIdx + 7);
        }
    }
        
    fs.writeFileSync(file, content, 'utf8');
    console.log(`Patched ${file}`);
});

File: patch_shell.py
import os

files_to_patch = [
    "cuti.html",
    "pelanggaran.html",
    "dashboard/index.html",
    "tip.html",
    "peminjaman.html",
    "peminjaman_admin.html",
    "karyawan.html",
    "users.html"
]

for file in files_to_patch:
    if not os.path.exists(file):
        print(f"File not found: {file}")
        continue
    
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    prefix = "../" if "/" in file else ""
    base_dir = ".." if "/" in file else "."
    
    # 1. Inject CSS links
    css_links = f'  <link rel="stylesheet" href="{prefix}shared/design-tokens.css">\n  <link rel="stylesheet" href="{prefix}shared/shell.css">\n  <style>'
    
    if '<link rel="stylesheet" href="' + prefix + 'shared/design-tokens.css">' not in content:
        content = content.replace('<style>', css_links, 1)
        
    # 2. Inject shell.js
    js_link = f'  <script src="{prefix}shared/shell.js" data-base="{base_dir}"></script>\n</body>'
    
    if '<script src="' + prefix + 'shared/shell.js"' not in content:
        # handle different spacing/newlines before </body>
        content = content.replace('</body>', js_link)
        
    with open(file, 'w', encoding='utf-8') as f:
        f.write(content)
        
    print(f"Patched {file}")
